homeassistant/components/light/philipslight.py

```diff
@@ -0,0 +1,215 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Philips Light'
+PLATFORM = 'philipslight'
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): vol.All(str, vol.Length(min=32, max=32)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+REQUIREMENTS = ['python-mirobo==0.1.3']
+
+# The light does not accept cct values < 1
+CCT_MIN = 1
+CCT_MAX = 100
+
+SUCCESS = ['ok']
+
+
+# pylint: disable=unused-argument
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the light from config."""
+    from mirobo import Ceil
+    if PLATFORM not in hass.data:
+        hass.data[PLATFORM] = {}
+
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    token = config.get(CONF_TOKEN)
+
+    _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
+    light = Ceil(host, token)
+
+    philips_light = PhilipsLight(name, light)
+    hass.data[PLATFORM][host] = philips_light
+
+    async_add_devices([philips_light], update_before_add=True)
+
+
+class PhilipsLight(Light):
+    """Representation of a Philips Light."""
+
+    def __init__(self, name, light):
+        """Initialize the light device."""
+        self._name = name
+
+        self._brightness = 180
+        self._color_temp = None
+
+        self._light = light
+        self._state = None
+
+    @property
+    def should_poll(self):
+        """Poll the light."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return true when state is known."""
+        return self._state is not None
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return self._brightness
+
+    @property
+    def color_temp(self):
+        """Return the color temperature."""
+        return self._color_temp
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        return math.floor(kelvin_to_mired(5700))
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        return math.floor(kelvin_to_mired(3000))
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP
+
+    @asyncio.coroutine
+    def _try_command(self, mask_error, func, *args, **kwargs):
+        """Call a light command handling error messages."""
+        from mirobo import DeviceException
+        try:
+            result = yield from self.hass.async_add_job(
+                partial(func, *args, **kwargs))
```

Comment:
All that run in a thread and not inside loop. You should direct yield from function if that is a coroutine or call that without `async_add_job` for run inside Loop.